src/AWS.Deploy.Orchestration/RecipeHandler.cs

```diff
@@ -225,5 +236,31 @@ private bool ContainsRecipeFile(string directoryPath)
 
             return recipeFilePaths.All(filePath => Path.GetFileNameWithoutExtension(filePath).Equals(directoryName, StringComparison.Ordinal));
         }
+        
+        /// Creates an option setting item dependency tree that indicates
+        /// which option setting items need to be validated if a value update occurs.
+        /// The function recursively goes through all the settings and their children to build this tree.
+        /// This method also creates a Leaf Id which will help reference <see cref="OptionSettingItem"/>.
+        /// </summary>
+        private void BuildDependencyTree(RecipeDefinition recipe, List<OptionSettingItem> optionSettingItems, Dictionary<string, List<string>> dependencyTree, string parentLeafId = "")
+        {
+            foreach (var optionSettingItem in optionSettingItems)
+            {
+                optionSettingItem.FullyQualifiedId = $"{parentLeafId}{optionSettingItem.Id}";
```

Comment:
Scan the code change to remove the term `Leaf`. Since I was already pushy to rename the property lets not have the artifacts of 2 different naming patterns.

Reply:
Done